source/enum_set.h

```diff
@@ -78,68 +78,190 @@ class EnumSet {
   static constexpr size_t kBucketSize = sizeof(BucketType) * 8ULL;
 
  public:
+  class Iterator {
+   public:
+    typedef Iterator self_type;
+    typedef T value_type;
+    typedef T& reference;
+    typedef T* pointer;
+    typedef std::forward_iterator_tag iterator_category;
+    typedef size_t difference_type;
+
+    Iterator(const EnumSet* set, size_t bucketIndex, ElementType bucketOffset)
+        : set_(set), bucketIndex_(bucketIndex), bucketOffset_(bucketOffset) {}
```

Comment:
Could you make this constructor private or protected? I don't think code outside of the EnumSet class should be able to use it.

Reply:
Yes I can, I'm just a dumby-dumb-dumb who forgot `friend` exits.. Thanks 😀